jerry-core/parser/js/js-parser-module.c

```diff
@@ -342,20 +339,28 @@ parser_module_parse_export_clause (parser_context_t *context_p) /**< parser cont
       if (context_p->token.type != LEXER_LITERAL
           || context_p->token.lit_location.type != LEXER_IDENT_LITERAL)
       {
-        ecma_deref_ecma_string (local_name_p);
-        ecma_deref_ecma_string (export_name_p);
         parser_raise_error (context_p, PARSER_ERR_IDENTIFIER_EXPECTED);
       }
 
       lexer_construct_literal_object (context_p, &context_p->token.lit_location, LEXER_IDENT_LITERAL);
 
-      ecma_deref_ecma_string (local_name_p);
-      export_name_p = ecma_new_ecma_string_from_utf8 (context_p->lit_object.literal_p->u.char_p,
-                                                      context_p->lit_object.literal_p->prop.length);
+      export_name_index = context_p->lit_object.index;
 
       lexer_next_token (context_p);
     }
 
+    lexer_literal_t *literal_p = PARSER_GET_LITERAL (local_name_index);
+    local_name_p = ecma_new_ecma_string_from_utf8 (literal_p->u.char_p, literal_p->prop.length);
+    ecma_ref_ecma_string (local_name_p);
+    export_name_p = local_name_p;
```

Comment:
These last two lines could be moved into an `else` block after the condition below, so we don't ref/deref unnecessarily